qiling/os/memory.py

```diff
@@ -63,6 +63,29 @@ def __read_string(self, addr: int) -> str:
     def __write_string(self, addr: int, s: str, encoding: str):
         self.write(addr, bytes(s, encoding) + b'\x00')
 
+    def __getitem__(self, key: Union[slice, int]) -> bytearray:
+        if isinstance(key, slice):
+            start = key.start
+            stop = key.stop
+            step = key.step
+
+            if step is not None and step != 1:
+                # step != 1 means we have to do copy, don't allow it.
+                raise IndexError("Only support slicing continous memory")
+
+            return self.ql.mem.read(start, stop - start)
```

Comment:
Probably needs a check that `start` is not greater than `stop`, or we end up with a negative read size.
Perhaps something like `size = max(0, stop - start)` would do, but I don't remember how `read` behaves with a zero-length read size. Either way, the `bytearray` semantics defines a return value of an empty `bytearray` in such a case.

Reply:
If `size=0`, the buffer is `bytearray(b'')` and left untouched in `uc_mem_read`

Reply:
done